W-23030047 pdk unit 1 9 updates gr (#1089)
mulesoft/docs-gateway: pdk/1.9/modules/ROOT/pages/policies-pdk-unit.adoc

[source,toml]
----
[dev-dependencies]
pdk-unit = { version = "1.9" }
----

=== Enable pdk-unit for Non-PDK Policies

> PDK-Unit supports testing any policy built with https://github.com/proxy-wasm/proxy-wasm-rust-sdk[proxy-wasm-rust-sdk]. You can write and run unit tests for existing proxy-wasm policies without needing to migrate them to PDK first. Use this functionality to:

Reviewer: ```suggestion
PDK-Unit supports testing any policy built with https://github.com/proxy-wasm/proxy-wasm-rust-sdk[proxy-wasm-rust-sdk]. You can write and run unit tests for existing proxy-wasm policies without migrating them to PDK first. Use this functionality to:
```

* Validate behavior before migration: Test your legacy proxy-wasm policies to establish a behavioral baseline.
* Identify regressions during migration: Establish tests before migrating your policy to PDK to make sure behavioral differences surface immediately.
* Adopt PDK incrementally: Migrate your policies to PDK at your own pace while still performing unit testing.

To test non-PDK policies with pdk-unit:

. Make these edits to the custom policy's `Cargo.toml` file before compilation:
+

> * Replace the `pdk-proxy-wasm-stub` target instead of `proxy-wasm`.

Reviewer: ```suggestion
* Replace the `pdk-proxy-wasm-stub` target instead of using `proxy-wasm`.
```

+

> NOTE: Don't make this edit for wasm32 architecture.

Reviewer: ```suggestion
NOTE: Don't make this edit for `wasm32` architecture.
```
